### ISSUE TITLE
[Bug]: When using AutoHeight() in a scrollable body  if a slide is has overflow it causes infinite loop

### ISSUE THREAD

**Issue by shantk705** [davidjerleke/embla-carousel#954]
### Which variants of Embla Carousel are you using?

- [X] embla-carousel **(Core)**
- [X] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [X] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-fade
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

### Steps to reproduce

The bug occurs when the  the body of the page is scrollable and the carousel has a long slide which overflows

- First issue the carousel wont work with slide gesture  and when you use the scroll Snaps it does not transition 
- Second and main issue is that the Auto Height () option causes infinite loop

### Expected Behavior

that the carousel should  behave normally and no loop should occur

### Additional Context

I have also tried...

the only  way to stop the loop is to  make the body not scrollable and give overflow scroll to the carousel which is not a solution to my current issue

### What browsers are you seeing the problem on?

All browsers

### Version

8.1.7

### CodeSandbox

https://codesandbox.io/p/sandbox/embla-carousel-auto-height-7xszcr?file=%2Fpublic%2Findex.html

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues
- [X] I agree to follow this project's Contributing Guidelines for bug reports

**Comment by davidjerleke** [comment]
@shantk705 please complete your bug report with clear steps on how to reproduce the problem.

**Comment by shantk705** [comment]
@davidjerleke  hello, 

In the code Sandbox  i added an example of what's happening with me  when the page is scrollable and  one of the slides in the carousel is big enough to cause a scroll or overflow  if you navigate to other slides its causing an infinite loop .

Steps to reproduce :
- Open the Sandbox  and open inspect in the browser i left a console log in the code which will log every time the carousel is rendered 

- When you go to the slide that has 30 elements (has overflow) the loop will stop  but when you navigate back to other slides it will start again and  you will notice that you cant use the swipe gesture to navigate only clicking on the tab will take you to the slide (keep in mind that the animation is not working when this occurs)

**Comment by shantk705** [comment]
@davidjerleke  are you able to slide between the tabs in the sandbox i created also when you open  my sandbox  the infinite loop is already triggered without pressing anything  if you can please check it again 

Also i'm facing another issue where if  one of the slides has dynamic data (fetched from an api ) and after the data is fetched   it changes size the carousel is not automatically updating  , lets say i have a loader which is around 400 pixels once the data is fetched the slide is 700 pixels now (not a fixed size its based on the fetched data ) the  slide is staying 400 pixels im adding a use Effect to use the reInit() every time the data changes do you want me to create a sandbox for that too or is it clear enough?

**Comment by davidjerleke** [comment]
> @davidjerleke are you able to slide between the tabs in the sandbox i created also when you open my sandbox the infinite loop is already triggered without pressing anything if you can please check it again

@shantk705 yes l’m able to do so. It’s working as expected for me. No infinite loop on my side and I’m using a Mac.

> I just tested in on many devices its happening on windows devices on mac its not looping and working totally fine.

Could you try creating a setup that doesn’t use CodeSandbox locally on a machine where the problem is present and test if the infinite loop still happens? For example you can spin up a new Vite template in no time and use the exact same carousel code. Because Embla has no OS specific code so it surely sounds a bit odd.

> Also i'm facing another issue where if one of the slides has dynamic data (fetched from an api ) and after the data is fetched it changes size the carousel is not automatically updating

Automatically listening for height changes can cause problems because some animation libraries animate the height in steps and this would trigger many `reInits`. But you can use this approach in your case:

- https://github.com/davidjerleke/embla-carousel/issues/910#issuecomment-2180553408

**Comment by davidjerleke** [comment]
@shantk705 did you see [my response](https://github.com/davidjerleke/embla-carousel/issues/954#issuecomment-2262155899)?

**Comment by davidjerleke** [comment]
> [!IMPORTANT]  
> This issue is scheduled for deletion in two days due to lack of response.
